sdk/translation/azure-ai-translation-text/src/main/java/com/azure/ai/translation/text/TextTranslationClientBuilder.java

```diff
@@ -342,11 +361,19 @@ private HttpPipeline createHttpPipeline() {
         policies.add(new AddDatePolicy());
         if (tokenCredential != null) {
             policies.add(new BearerTokenAuthenticationPolicy(tokenCredential, DEFAULT_SCOPE));
+            if (this.region != null && this.resourceId != null) {
```

Comment:
What if one of them is null and the other is not? We should throw an error in this case as ignoring one of them silently will give the impression that region/resourceId is being used.

Reply:
Added check.